flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/ArrayType.java

```diff
@@ -42,6 +42,7 @@
 	public static final String FORMAT = "ARRAY<%s>";
 
 	private static final Set<String> INPUT_OUTPUT_CONVERSION = conversionSet(
+		List.class.getName(),
```

Comment:
Why add it here when we anyways have the explicit check for `List` later. Ahh, it's because in input conversions we allow subtypes but not for output conversions, right?

Reply:
exactly, this list is for both input and output